flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolImplTest.java

```diff
@@ -145,14 +133,13 @@ public void testAllocateSimpleSlot() throws Exception {
 
 			assertTrue(slotPool.offerSlot(taskManagerLocation, taskManagerGateway, slotOffer));
 
-			LogicalSlot slot = future.get(1, TimeUnit.SECONDS);
+			final PhysicalSlot physicalSlot = future.get(1, TimeUnit.SECONDS);
 			assertTrue(future.isDone());
-			assertTrue(slot.isAlive());
-			assertEquals(taskManagerLocation, slot.getTaskManagerLocation());
+			assertEquals(taskManagerLocation, physicalSlot.getTaskManagerLocation());
+			assertEquals(slotRequest.getAllocationId(), physicalSlot.getAllocationId());
 		}
 	}
 
-	@Nonnull
 	private SlotPoolImpl createSlotPoolImpl() {
```

Comment:
I'd prefer to move this method to be below `createAndSetUpSlotPool()` because it is first and only invoked there.

Reply:
It's fine to not do this because in a later commit you will remove it.
But then I think you also do not need to remove `@Nonnull` in this commit.